### PR TITLE
[4] Make behindproxy_postinstall_action compatible with Joomla 4 and remove FTP layer

### PR DESCRIPTION
**Pull request by PhilETaylor**
Fixes merge forgetfulness, and refactors to make compatible with Joomla 4 class names and libs.

Also removes useless FTP checks now that the FTP layer has gone.

### CODE REVIEW

File: administrator/components/com_admin/postinstall/behindproxy.php [hunk 1]
```diff
@@ -57,34 +60,30 @@ function behindproxy_postinstall_action()
...
+		return;
 	}
 
 	// Attempt to write the configuration file as a PHP class named JConfig.
 	$configuration = $config->toString('PHP', array('class' => 'JConfig', 'closingtag' => false));
 
 	if (!File::write($file, $configuration))
 	{
-		JFactory::getApplication()->enqueueMessage(JText::_('COM_CONFIG_ERROR_WRITE_FAILED'), 'error');
+		JFactory::getApplication()->enqueueMessage(Text::_('COM_CONFIG_ERROR_WRITE_FAILED'), 'error');
```
Comment (Quy) [inline administrator/components/com_admin/postinstall/behindproxy.php]:
```suggestion
		Factory::getApplication()->enqueueMessage(Text::_('COM_CONFIG_ERROR_WRITE_FAILED'), 'error');
```

**Comment by brianteeman** [comment]
even though there was already a pr for this I will close mine

**Comment by PhilETaylor** [comment]
Really? Sorry I thought the one you did would have been merged by now and that this was still left. 

I guess I just got used to PRs being merged the same day like they were for a while. 

I'm happy to close this and leave yours open :-) up to you

**Comment by brianteeman** [comment]
already closed mine

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f97b4e017bcad0f84839b95c82ca1bde512aafcf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34018">issues.joomla.org/tracker/joomla-cms/34018</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on f97b4e017bcad0f84839b95c82ca1bde512aafcf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34018">issues.joomla.org/tracker/joomla-cms/34018</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34018">issues.joomla.org/tracker/joomla-cms/34018</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34018">issues.joomla.org/tracker/joomla-cms/34018</a>.</sub>

**Pull request merged**